src/morale.cpp

```diff
@@ -0,0 +1,163 @@
+            { _( "Good Feeling" ) },
+            { _( "Supported" ) },
+            { _( "Looked at photos" ) },
+
+            { _( "Nicotine Craving" ) },
+            { _( "Caffeine Craving" ) },
+            { _( "Alcohol Craving" ) },
+            { _( "Opiate Craving" ) },
+            { _( "Speed Craving" ) },
+            { _( "Cocaine Craving" ) },
+            { _( "Crack Cocaine Craving" ) },
+            { _( "Mutagen Craving" ) },
+            { _( "Diazepam Craving" ) },
+            { _( "Marloss Craving" ) },
+
+            { _( "Disliked %i" ) },
+            { _( "Ate Human Flesh" ) },
+            { _( "Ate Meat" ) },
+            { _( "Ate Vegetables" ) },
+            { _( "Ate Fruit" ) },
+            { _( "Lactose Intolerance" ) },
+            { _( "Ate Junk Food" ) },
+            { _( "Wheat Allergy" ) },
+            { _( "Ate Indigestible Food" ) },
+            { _( "Wet" ) },
+            { _( "Dried Off" ) },
+            { _( "Cold" ) },
+            { _( "Hot" ) },
+            { _( "Bad Feeling" ) },
+            { _( "Killed Innocent" ) },
+            { _( "Killed Friend" ) },
+            { _( "Guilty about Killing" ) },
+            { _( "Guilty about Mutilating Corpse" ) },
+            { _( "Fey Mutation" ) },
+            { _( "Chimerical Mutation" ) },
+            { _( "Mutation" ) },
+
+            { _( "Moodswing" ) },
+            { _( "Read %i" ) },
+            { _( "Got comfy" ) },
+
+            { _( "Heard Disturbing Scream" ) },
+
+            { _( "Masochism" ) },
+            { _( "Hoarder" ) },
+            { _( "Stylish" ) },
+            { _( "Optimist" ) },
+            { _( "Bad Tempered" ) },
+            //~ You really don't like wearing the Uncomfy Gear
+            { _( "Uncomfy Gear" ) },
+            { _( "Found kitten <3" ) },
+
+            { _( "Got a Haircut" ) },
+            { _( "Freshly Shaven" ) },
+        } };
+
+        if( static_cast<size_t>( id ) >= morale_data.size() ) {
+            debugmsg( "invalid morale type: %d", id );
+            return morale_data[0];
+        }
+
+        return morale_data[id];
+    }
+} // namespace
+
+std::string morale_point::name() const
+{
+    // Start with the morale type's description.
+    std::string ret = get_morale_data( type );
+
+    // Get the name of the referenced item (if any).
+    std::string item_name = "";
+    if( item_type != nullptr ) {
+        item_name = item_type->nname( 1 );
+    }
+
+    // Replace each instance of %i with the item's name.
+    size_t it = ret.find( "%i" );
+    while( it != std::string::npos ) {
+        ret.replace( it, 2, item_name );
+        it = ret.find( "%i" );
+    }
+
+    return ret;
+}
+
+void morale_point::add( int bonus, int max_bonus, int duration, int decay_start, bool cap_existing )
+{
+    // If we're capping the existing effect, we can use the new duration
+    // and decay start.
+    if( cap_existing ) {
+        this->duration = duration;
```

Comment:
> Getters then can be shortened (i.e. just type() instead of get_type()). It's not written anywhere directly and various styles are equally frequent in the code, so I hesitated which one to choose here.

I think we use `get_[val]` more than just `val` getters, at least for pure wrapper getters that do nothing on their own.